debian/ubuntu-advantage-tools.postinst

```diff
@@ -2,20 +2,22 @@
 
 set -e
 
-. /etc/os-release  # For VERSION_ID
-
-# Since UBUNTU_CODENAME isn't on trusty set it set a default if unknown
-if [ "" = "${UBUNTU_CODENAME}" ]; then
-   case "$VERSION_ID" in
-       14.04) UBUNTU_CODENAME="trusty";;
-       *) UBUNTU_CODENAME="NO-UBUNTU_CODENAME-$VERSION_ID";;
-   esac
-fi
+. /etc/os-release  # For VERSION_ID and UBUNTU_CODENAME
 
 # Needed even if this script doesn't call debconf, see:
 # https://lintian.debian.org/tags/postinst-does-not-load-confmodule.html
+# Note: this may re-exec the postinst script.
 . /usr/share/debconf/confmodule
 
+if [ -z "${VERSION_ID}" ]; then
+    echo "Warning: missing VERSION_ID in /etc/os-release" >&2
+    VERSION_ID="NO-VERSION_ID"
+fi
+if [ -z "${UBUNTU_CODENAME}" ]; then
+    echo "Warning: missing UBUNTU_CODENAME in /etc/os-release" >&2
+    UBUNTU_CODENAME="NO-UBUNTU_CODENAME-$VERSION_ID"
+fi
+
```

Comment:
@blackboxsw Please take a look. This essentially does the same as what Renan had, but also warns about the situation. I think this is a good middleground, because an early exit might prevent other important postinst tasks from running that don't require these variables. LMK what you think